src/redux/groups/groupsSlice.ts

```diff
@@ -1,38 +1,62 @@
+import { TreeNodeData } from '@mantine/core';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
-import { Groups } from '@/types/types';
+import { Group } from '@/types/types';
+
+export type Groups = {
+  [key: string]: Group;
+};
+
+export type CustomGroupOrdering = {
+  [key: string]: string[]; // group paths
```

Comment:
Think I commented on a similar comment somewhere else, but I would like to discuss this!  Have a hard time understanding why it would be necessary. 

The object should not contain any keys for which the value is undefined, and non-existing keys should already be handled by the object type? Please explain to me how it works if I have misunderstood :)